### ISSUE TITLE
Crash on opening building information

### ISSUE THREAD

**Issue by juj** [nicymike/serflings#73]
Tried a multiplayer game with a friend. Had a nice time, though game kept crashing every ~20 minutes. One crash callstack:

![image](https://user-images.githubusercontent.com/225351/179581120-46235707-06d6-4a6a-a05b-7d0684e8cb99.png)

**Comment by nicymike** [comment]
On which version of the game did you play?

Better post the text because I need it to convert names.
```
java.lang.IllegalArgumentException: Invalid index 869
  at net.simpleguide.b.a.a.g.a(net.simpleguide.Main:54)
  at net.simpleguide.b.a.k.a.z.a(net.simpleguide.Main:13)
  at net.simpleguide.b.a.k.a.D.a(net.simpleguide.Main:129)
  at net.simpleguide.b.a.k.a.D.n(net.simpleguide.Main:1952)
  at net.simpleguide.b.a.k.a.D.b(net.simpleguide.Main:3240)
  at net.simpleguide.a.a.e.f.a(net.simpleguide.Main:86)
  at net.simpleguide.a.a.e.a.a.b(net.simpleguide.Main:166)
  at net.simpleguide.a.a.e.a.a(net.simpleguide.Main:54)
  at net.simpleguide.a.a.a.run(net.simpleguide.Main:1499)
  at java.lang.Thread.run(Thread.java:748)
```

**Comment by juj** [comment]
I am playing the latest version that was available for download: https://download.simpleguide.net/serflings/serflings-1.1.0-win64.zip

Is the source code available somewhere?

**Comment by nicymike** [comment]
Sorry, no source code available because of legal and copyright issues.

The crash happens if you have a window open displaying an invalid building.
Any ideas what happened before the crash?

At what game speed did you play? Did you conquer a building before the crash?
Do you have a save game available to test?

**Comment by juj** [comment]
Sorry, unfortunately no save game. This occurred on my friend's client - he mentioned that he had middle-clicked on a building during the game.

We did not change the game speed (did not know there is an option to do so). If I remember correctly, there had not yet been any fights that would have razed/conquered buildings when that crash happened.

**Comment by nicymike** [comment]
Game speed may be changed only by using keyboard shortcuts. Check out the website for all available keys.

Did you play a coop game or a regular game with two players?

**Comment by juj** [comment]
It was a regular game 1v1, though our borders were yet to make a combat contact. No AI opponents.

**Comment by nicymike** [comment]
Thank you for the information. I know where the crash occurs and I guess I have a workaround but I am still checking why this crash happens in the first place. Somehow it is possible to open the building information window (middle click on building) with an invalid building selected.

**Comment by nicymike** [comment]
I found the actual bug and I am surprised that it was not discovered before. Thank you for reporting.

The crash happens if you open the building information for a "normal" building (no mine, no military, no stock) by using the special or middle click and then open the window for a "special" building (mine, military, stock). I can reproduce the crash every time I am doing this, even in singleplayer.

I will work on a fix but until that close the window for building information every time before opening a new one. This should help you out in playing a game without crashes. :)

**Comment by nicymike** [comment]
Fixed in 1.1.1
